### ISSUE TITLE
Improve FiberId startTime's granularity from seconds to a fraction of second

### ISSUE THREAD

**Issue by landerlo** [zio/zio#6658]
The Fiber's start time could have a finer time granularity without increasing `startTimeSeconds: Int` footprint. A trivial solution is to make it a delta from the JVM startup time. Considering we already carry an Int for this information it would be nice to always have access to more accurate startup times, especially relevant for short timed fibers when monitoring.

The tradeoff would be a limited JVM instance lifetime before `startTime` becomes inaccurate due to overflow, depending on the selected granularity. Granularity (millisecond resolution) to max JVM instance life relation goes as, 16ms  ->  ~400 days,  32ms -> ~1600d or the most conservative, 64ms -> ~9 years of max life.

If there's appetite to make this change I can make a PR for this.

**Comment by adamgfraser** [comment]
That makes sense to me. The main downside I see is that it makes serialization logic more complex.

**Comment by landerlo** [comment]
Yes, readObject and writeObject would need to be overriden to support Serialization and the serialized startTime needs to be an absolute value, so either serialized binary footprint is increased to serialize a `Long` with `currentMillis` or we serialize an `Int` with `currentSeconds` thus losing the subsecond precission when recreating the id.

**Comment by hearnadam** [comment]
Is this still an issue in ZIO 2.x @landerlo ?
